sdk/eventhubs/azure-messaging-eventhubs/test/ut/processor_test.cpp

```diff
@@ -217,8 +217,8 @@ namespace Azure { namespace Messaging { namespace EventHubs { namespace Test {
 
     void TestPartitionAcquisition(Models::ProcessorStrategy processorStrategy)
     {
-      Azure::Core::Context context
-        = Azure::Core::Context{} .WithDeadline(Azure::DateTime::clock::now() + std::chrono::minutes(5));
+      Azure::Core::Context context = Azure::Core::Context{}.WithDeadline(
```

Comment:
It would look so much nicer if it was `Azure::Core::Context.WithDeadline(....)` rather than constructing an empty to call WithDeadline to then return a 2nd constructed context.

Could `Context` have or be a factory?

Reply:
I completely agree with you that it would look better. That's why I added the constructor form of the method.

But Context is a type, and the only way to make it work the way you would like it to work would be for WithDeadline to be a static method. And unfortunately, you can't have static and class methods with the same name (your solution was exactly my first attempt at fixing the issue). 

The reason I changed it to the current form is that it allows storage, identity, and eventhubs to be released *without* taking a dependency on the new core release.

The construct as written will work correctly with older core releases AND with the current core release. So unless we want to block future releases of storage, identity and eventhubs until we release a public core release with the context changes, this is the best that we can get, even though it's an incredibly awkward construct.